modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAngularClientCodegen.java

```diff
@@ -486,7 +505,44 @@ private String getApiFilenameFromClassname(String classname) {
 
     private String getModelnameFromModelFilename(String filename) {
         String name = filename.substring((modelPackage() + "/").length());
-        return camelize(name);
+        // Remove the file suffix and add the class suffix.
+        // This is needed because the model file suffix might not be the same as
+        // the model suffix.
+        if (modelFileSuffix.length() > 0) {
+            name = name.substring(0, name.length() - modelFileSuffix.length());
+        }
+        return camelize(name) + modelSuffix;
+    }
+
+    @Override
+    public String toModelName(String name) {
+        String modelName = super.toModelName(name);
+        if (modelSuffix.length() == 0 || modelName.endsWith(modelSuffix)) {
+            return modelName;
+        }
+        return modelName + modelSuffix;
+    }
+
+    private String removeModelSuffixIfNecessary(String name) {
+        if (modelSuffix.length() == 0 || !name.endsWith(modelSuffix)) {
+            return name;
+        }
+        return name.substring(0, name.length() - modelSuffix.length());
     }
 
+    private void validateFileSuffixArgument(String argument, String value) {
```

Comment:
please add a method description. describe what `argument` and `value` are or rename them to better understand their meaning